angular-with-directus/how to integrate directus with Angular.md

```diff
@@ -0,0 +1,449 @@
+[Angular](https://angular.dev/) is a popular front-end web framework. In this tutorial, you will use the framework to implement the front-end for the Directus headless CMS. You will implement a blog that loads blog posts dynamically and also serves global metadata.
+
+## Prerequisites
+- Some knowledge of TypeScript and [Angular](https://angular.io/)
+- A Directus project. Follow the [Quickstart guide](https://docs.directus.io/getting-started/quickstart.html) to create one.
+- [Node.js](https://nodejs.org/en/download) and a development environment of your choice
+- Install the Angular CLI - use the Angular [guide](https://angular.io/guide/setup-local) to achieve this. 
+
+:::info Compatibility
+
+Note that Angular and TypeScript versions must be compatible. Since the SDK requires a minimum TypeScript version of 5.0, you need to use Angular version 17 for your project.
+
+:::
+
+## Initialize Project
+To create a new Angular project, use the following command.
+
+```bash
+ng new directus-with-angular
+? Which stylesheet format would you like to use? CSS
+? Do you want to enable Server-Side Rendering (SSR) and Static Site Generation (SSG/Prerendering)? (y/N) No
+```
+Next, run the following command to install the Directus SDK:
+
+```bash
+npm install @directus/sdk
+```
+Once the project has been created, open it in your code editor and replace the code in the `src/app/app.component.html` file with the following:
+
+```html
+<router-outlet/>
+```
+Angular will dynamically fill the [RouterOutlet](https://angular.io/api/router/RouterOutlet) placeholder based on the current router state.
+
+You should also disable strict checking in `./tsconfig.json` file under `compilerOptions`.
+
+```json
+"strict": false
+```
+
+Navigate to your project directory in a terminal and start the development server at `http://localhost:4200`:
+
+```bash
+ng serve
+```
+## Create an instance of Directus SDK
+For every Directus model that you define, you need to create a TypeScript type for that model. The type will help to map the JSON data to TypeScript objects.
+
+In addition, you should expose an instance of the Directus SDK that you will use to make different requests to the Directus CMS.
+
+In your project, create a file named `./directus.ts` with the following code:
+
+```ts
+import {createDirectus, rest} from "@directus/sdk";
+
+type Global = {
+  id: number;
+  title: string;
+  description: string;
+}
+
+type Author = {
+  id: number;
+  name: string;
+}
+
+type Page = {
+  slug: string;
+  title: string;
+  content: string;
+}
+
+type Post = {
+  id: number;
+  image: string;
+  title: string;
+  content: string;
+  author: Author;
+  published_date: string;
+}
+
+type Schema = {
+  global: Global;
+  posts: Post[];
+  pages: Page[];
+}
+
+const directus =
+  createDirectus<Schema>("YOUR_DIRECTUS_URL")
+    .with(rest());
+
+export {directus, Global, Post, Page}
+```
```

Comment:
Add a sentence or two here to say each type represents one collection we will create later in the tutorial, and each property is a field. As `global` is a singleton, we do not define it as an array in the Schema.

Reply:
Okay.